oci/mounts_freebsd.go

```diff
@@ -0,0 +1,56 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package oci
+
+import (
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+)
+
+func defaultMounts() []specs.Mount {
+	return []specs.Mount{
+		{
+			Destination: "/proc",
+			Type:        "procfs",
+			Source:      "proc",
+			Options:     []string{"nosuid", "noexec"},
+		},
+		{
+			Destination: "/dev",
+			Type:        "devfs",
+			Source:      "devfs",
+			Options:     []string{},
+		},
+		{
+			Destination: "/dev/fd",
+			Type:        "fdescfs",
+			Source:      "fdescfs",
+			Options:     []string{},
+		},
+		{
+			Destination: "/dev/mqueue",
+			Type:        "mqueue",
+			Source:      "mqueue",
+			Options:     []string{"nosuid", "noexec"},
+		},
+		{
+			Destination: "/dev/shm",
+			Type:        "tmpfs",
+			Source:      "shm",
+			Options:     []string{"nosuid", "noexec", "mode=1777"},
+		},
```

Comment:
I admit to being very much a newbie with FreeBSD, but I'm wondering whether these are the right defaults.  On my FreeBSD host (outside a jail) I only have `devfs` mounted at `/dev`.  `/dev/fd` appears to be provided by `devfs`, and the same is true in the jails that I've set up with [runj](https://github.com/samuelkarp/runj/blob/de37a70b4b23e470b4d4c7bab4c0aa648b8eb87d/jail/conf.go#L18).  I don't have `/dev/shm` or `/dev/mqueue` at all on my host either.

```
$ mount
zroot/ROOT/default on / (zfs, local, noatime, nfsv4acls)
devfs on /dev (devfs, local, multilabel)
zroot/tmp on /tmp (zfs, local, noatime, nosuid, nfsv4acls)
zroot/usr/home on /usr/home (zfs, local, noatime, nfsv4acls)
zroot/var/log on /var/log (zfs, local, noatime, noexec, nosuid, nfsv4acls)
zroot on /zroot (zfs, local, noatime, nfsv4acls)
zroot/usr/ports on /usr/ports (zfs, local, noatime, nosuid, nfsv4acls)
zroot/var/audit on /var/audit (zfs, local, noatime, noexec, nosuid, nfsv4acls)
zroot/usr/src on /usr/src (zfs, local, noatime, nfsv4acls)
zroot/var/crash on /var/crash (zfs, local, noatime, noexec, nosuid, nfsv4acls)
zroot/var/mail on /var/mail (zfs, local, nfsv4acls)
zroot/var/tmp on /var/tmp (zfs, local, noatime, nosuid, nfsv4acls)
zroot/containerd on /var/lib/containerd/io.containerd.snapshotter.v1.zfs (zfs, local, noatime, nfsv4acls)
zroot/containerd/44 on /run/containerd/io.containerd.runtime.v2.task/default/test2/rootfs (zfs, local, noatime, nfsv4acls)
devfs on /run/containerd/io.containerd.runtime.v2.task/default/test2/rootfs/dev (devfs, local, multilabel)
```

Reply:
The man page for [`fdescfs`](https://www.freebsd.org/cgi/man.cgi?query=fdescfs&sektion=5&manpath=freebsd-release-ports) answers at least the question of `/dev/fd` being provided by `devfs`, but I'd still like to understand more about whether these are the appropriate defaults.

Reply:
Looking at my jails (setup via cbsd), I have fdescfs mounted for all of them, the other mounts are indeed not present. 

/dev/shm is a Linuxism, and I should remove that, my first intention was to make sure mount doesn't explode when it tries to mount filesystems that are not native to FreeBSD. /dev/mqueue is also save to remove I'd say, it doesn't seem to be mounted anywhere either.

Reply:
I'll fixup tonight.

Reply:
> Looking at my jails (setup via cbsd), I have fdescfs mounted for all of them, the other mounts are indeed not present.

Should fdescfs be present by default though?  It doesn't seem to be present in the FreeBSD host I've been using that I installed without much customization.

Reply:
According to https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=216985 and [FreeBSD's ports](https://github.com/freebsd/freebsd-ports/search?q=%22mounted+on+%2Fdev%2Ffd%22), bash and OpenJDK need fdescfs.

So questions are;
-  Do we want to provide a minimal set or good enough default to run common programs?
- Are bash and/or OpenJDK considered "common" to mount fdescfs by default?

Reply:
bash and OpenJDK needing fdescfs sounds like a good reason to mount it by default to me.